turbo/jsonrpc/receipts/receipts_generator.go

```diff
@@ -106,38 +107,34 @@ func (g *Generator) addToCache(header *types.Header, receipts types.Receipts) {
 	g.receiptsCache.Add(header.Hash(), receipts.Copy()) // .Copy() helps pprof to attribute memory to cache - instead of evm (where it was allocated).
 }
 
-func (g *Generator) GetReceipt(ctx context.Context, cfg *chain.Config, tx kv.Tx, block *types.Block, index int, optimize bool) (*types.Receipt, error) {
+func (g *Generator) GetReceipt(ctx context.Context, cfg *chain.Config, tx kv.TemporalTx, block *types.Block, index int, txNum uint64) (*types.Receipt, error) {
 	if receipts, ok := g.receiptsCache.Get(block.Hash()); ok && len(receipts) > index {
```

Comment:
do we need one more cache for this method? (with txn granularity) - if yes, probably it's for another task

Reply:
I think yes